File: lib/enums/question_type.dart
enum QuestionType {
  multipleChoice,
  trueFalse,
  code,
  practical;

  String get label {
    switch (this) {
      case QuestionType.multipleChoice:
        return 'Choix multiple';
      case QuestionType.trueFalse:
        return 'Vrai/Faux';
      case QuestionType.code:
        return 'Code';
      case QuestionType.practical:
        return 'Pratique';
    }
  }

  static QuestionType parseType(String value) {
    switch (value.toLowerCase()) {
      case 'multiplechoice':
      case 'multiple_choice':
      case 'choix multiple':
        return QuestionType.multipleChoice;
      case 'truefalse':
      case 'true_false':
      case 'vrai/faux':
        return QuestionType.trueFalse;
      case 'code':
        return QuestionType.code;
      case 'pratique':
      case 'practical':
        return QuestionType.practical;
      default:
        return QuestionType.multipleChoice;
    }
  }
}

File: app/Enums/QuestionType.php
<?php

namespace App\Enums;

enum QuestionType: string
{
    case MULTIPLE_CHOICE = 'multiple_choice';
    case TRUE_FALSE = 'true_false';
    case CODE = 'code';
    case PRATICAL = 'practical';

    public function label(): string
    {
        return match($this) {
            self::MULTIPLE_CHOICE => "Question à choix multiples",
            self::TRUE_FALSE => "Vraie ou faux",
            self::CODE => "Code",
            self::PRATICAL => "Pratique"
        };
    }

    public static function values(): array
    {
        return array_column(self::cases(), 'value');
    }
}
